config/config.go

```diff
@@ -73,18 +73,12 @@ func (config *Config) Validate(path string) ([]string, error) {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "config_params[mode]")
 	}
 
-	if config.CloudStoryEnabled {
-		if config.ExistingMap == "" {
-			return nil, utils.NewConfigValidationFieldRequiredError(path, "existing_map")
-		}
```

Comment:
reverting to use `data_dir` for initial config changes

Reply:
remind me  again why we are doing this?

Reply:
If we use the `existing_map` field that will come in as part of cloud story, we will break all of the save internal StateLogic. By doing this we can split up the config changes as follows:

1. add `enable_slam` & use it to determine slam mode if cloud story is enabled
2. stop using `map_rate_sec` if clpud story is enabled and stop saving the internal state to `data_dir`
3. add in `existing_map` and use it if cloud story is enabled to get the existing map instead of `data_dir`

Does that make sense?

Reply:
what is `enable_slam`?

Reply:
a typo - I meant `enable_mapping` fixed

Reply:
got it, this makes sense